### PR TITLE
Deserialize arrays and vectors.

### LINKED ISSUE #47: How do I get multiple return values from the contract function?
There is a function of such a contract.
`function getValues() public constant returns (uint[], byte[64][], address[])`

I wrote the Rust code to receive the value, but this is an error.
```
let result = contract.query("getValues", (), None, Options::default());
let values: Vec<Token> = result.wait().unwrap();
```
```
error[E0277]: the trait bound `std::vec::Vec<ethabi::Token>: web3::contract::tokens::Tokenizable` is not satisfied
```

It worked when you made changes to crate, but I think this is a bad way.

`use ethabi::Token;` -> `use web3::contract::tokens::Token;`

> rust-web3/src/contract/tokens.rs:3
`use ethabi::Token;` -> `pub use ethabi::Token;`

> ethabi/src/token/token.rs:7
`#[derive(Debug, PartialEq, Clone)]` -> `#[derive(Debug, PartialEq, Clone, Deserialize)]`

How is the best way to do it?
Thank you.

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #47

### CODE REVIEW

File: src/contract/tokens.rs [hunk 1]
```diff
@@ -1,42 +1,76 @@
...
+  fn from_tokens(mut tokens: Vec<Token>) -> Result<Self, Error> {
+    if tokens.len() != 1 {
+      return Err(Error::InvalidOutputType(format!("Expected single element, got a list: {:?}", tokens)));
+    }
+    Self::from_token(tokens.drain(..).next().expect("At least one element in vector; qed"))
+  }
+}
+
+macro_rules! impl_output {
+  ($num: expr, $( $ty: ident : $no: tt, )+) => {
```
Comment (rphmeier) [inline src/contract/tokens.rs]: is $no used anymore?

Comment (tomusdrw) [inline src/contract/tokens.rs]: Good catch. Updated.

**Comment by tomusdrw** [comment]
What about `Detokenize` and `Tokenize`?

**Comment by debris** [comment]
sounds about right

**Pull request merged**